pkg/apis/events/v1alpha1/storage_defaults.go

```diff
@@ -34,4 +35,13 @@ func (gs *StorageSpec) SetDefaults(ctx context.Context) {
 	if equality.Semantic.DeepEqual(gs.GCSSecret, &corev1.SecretKeySelector{}) {
 		gs.GCSSecret = *v1alpha1.DefaultGoogleCloudSecretSelector()
 	}
+
+	if gs.Project == "" {
```

Comment:
This will not work in the operations world, the webhook will not have the secret that is required to call the metadata server I think

Reply:
Switched to just doing it in the reconciler.